app/src/sharedTest/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentTest.kt

```diff
@@ -526,6 +322,52 @@ class TopicLessonsFragmentTest {
     }
   }
 
+  private fun createTopicActivityIntent(internalProfileId: Int, topicId: String): Intent {
+    return TopicActivity.createTopicActivityIntent(
+      ApplicationProvider.getApplicationContext(),
+      internalProfileId,
+      topicId
+    )
+  }
+
+  private fun clickLessonTab() {
+    testCoroutineDispatchers.runCurrent()
+    onView(
+      allOf(
+        withText(TopicTab.getTabForPosition(1).name),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    testCoroutineDispatchers.runCurrent()
+  }
+
+  private fun clickStoryItem(position: Int, targetViewId: Int) {
+    onView(
+      atPositionOnView(
+        R.id.story_summary_recycler_view,
+        position,
+        targetViewId
+      )
+    ).perform(click())
+  }
+
+  private fun scrollToPosition(position: Int) {
```

Comment:
Should we still be using this per your earlier comment? I'm concerned about having 2 different scrolling solutions in Oppia Android--we ought to consolidate to a single one.

Reply:
So the flow is:
 - the clicking of a drop-down icon which expands the story card and shows chapter. 
 - We had already expand the first item
 - Now, when we try to scroll over the second item and perform a click, the click doesn't happen because the view is not visible completely **(The reason I think here is maybe the regular scrolling is not scrolling the item on a completely visible state on the screen. )**
 - Using `ViewActions.scrollTo()` does the hack here. 

`ViewActions.scrollTo()` work on the basis of View, which works on visible bounds under the hood - `view.getDrawingRect(rect);`
whereas
`RecyclerViewActions.scrollToPosition` works on the basis of item position.

Also, under the hood, `RecyclerViewActions` using `import android.support.v7.widget.RecyclerView;`.

Reply:
Adding @rt4914 for future reference

Reply:
Ah, in that case can you:
1. Introduce a new testing module file that provides scrolling actions
2. Wrap both of the two solutions behind two methods in the new class
3. Add documentation capturing the context above (the goal here is to provide sufficient context for developers to be able to decide between using one versus the other)
4. Update the references in this PR to use the new utilities
5. File a starter issue to update all other references in the codebase with the new utilities

Given how complex this is, I think we need to centrally manage scrolling so that we can be really deliberate about when to use option A or B. Long-term we can work to consolidate these, but we should avoid hiding context that requires esoteric knowledge (otherwise someone's just going to get confused in the future when they ask the same question & try digging into it).

Reply:
https://github.com/oppia/oppia-android/issues/2209
Split this work into another issue and will continue this on parallel side.

Reply:
Sg, thanks.